build_tools/bazel_to_cmake/bazel_to_cmake.py

```diff
@@ -176,23 +183,27 @@ def convert_directory(directory_path, write_files, allow_partial_conversion,
 
   old_lines = []
   preserved_footer_lines = ["\n" + PRESERVE_TAG + "\n"]
-  if os.path.isfile(cmakelists_file_path):
-    found_autogeneration_tag = False
-    found_preserve_tag = False
-    with open(cmakelists_file_path) as f:
-      old_lines = f.readlines()
-
-    for line in old_lines:
-      if not found_autogeneration_tag and autogeneration_tag in line:
-        found_autogeneration_tag = True
-      if not found_preserve_tag and PRESERVE_TAG in line:
-        found_preserve_tag = True
-      elif found_preserve_tag:
-        preserved_footer_lines.append(line)
-    if not found_autogeneration_tag:
-      if verbosity >= 1:
-        log(f"Skipped. Did not find autogeneration line.", indent=2)
-      return Status.SKIPPED
+  # Only process if there is a CMake file.
```

Comment:
Ah, ok. I was having trouble following the intent -- I think because I've never actually seen that feature work (I recall always needing to add a CMakeLists manually with the right header).

I don't disagree with followups to avoid the arbitrary (appearing) suffixes. As you know, these have magical properties in the Google monorepo, and it would be good to not have that magic leaking to the OSS repo.

Reply:
Reverted and added a check to skip if "SKIP: bazel_to_cmake" and annotated the offending file.

Reply:
Hmmm that shouldn't have been necessary, or at least not since the last time this was reworked. In fact, if you add a blank CMakeLists.txt it will cause the directory to get skipped since it has a CMakeLists.txt file without the magic autogenerated header